asio/src/examples/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+# Minimum supported CMake version
+cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
+
+# =================================================
+project(asio_examples VERSION 1.13.0 LANGUAGES CXX)
+# =================================================
+
+# Options
+set(ASIO_BUILD_EXAMPLES_CPP11 ON CACHE BOOL "Build asio C++11 examples")
+set(ASIO_BUILD_EXAMPLES_CPP14 ON CACHE BOOL "Build asio C++14 examples")
+if(NOT CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  set(ASIO_BUILD_EXAMPLES_CPP17 ON CACHE BOOL "Build asio C++17 examples")
+endif()
+
+# Check asio targets
+if(TARGET standalone)
+  add_library(asio::standalone ALIAS standalone)
+  set(ASIO_FOUND TRUE)
+endif()
+
+# Find asio package
+if(NOT ASIO_FOUND)
+  find_package(asio CONFIG REQUIRED)
+endif()
+
+# Installation configuration
+include(GNUInstallDirs)
+
+if(Boost_FOUND)
+  add_subdirectory(cpp03)
+else()
+  message(WARNING "examples/cpp03 needs boost!")
+endif()
```

Comment:
This check is unnecessary. Just include the `find_package(Boost REQUIRED)` call in the `cpp03` subproject. If boost is really needed, then the absence of boost is a hard error, not a warning.

Reply:
Maybe, but asI  said cpp03 is historic!